### PR TITLE
Controller and last service part

### PR DESCRIPTION
**Pull request by laurianed**

### CODE REVIEW

File: src/integration-test/java/org/ow2/proactive/catalog/rest/service/repository/QueryDslCatalogObjectRevisionRepositoryIntegrationTest.java [hunk 1]
```diff
@@ -66,14 +66,14 @@
 
 
 /**
- * Integration tests associated to {@link QueryDslWorkflowRevisionRepository}.
+ * Integration tests associated to {@link QueryDslCatalogObjectRevisionRepository}.
  * <p>
  * The idea is to start a Web application with a in-memory database that is
  * pre-allocated with several workflow revisions. Then, tests run a query
```
Comment (marcocast) [inline src/integration-test/java/org/ow2/proactive/catalog/rest/service/repository/QueryDslCatalogObjectRevisionRepositoryIntegrationTest.java]: workflow should be Object right?

File: src/integration-test/java/org/ow2/proactive/catalog/rest/service/repository/QueryDslCatalogObjectRevisionRepositoryIntegrationTest.java [hunk 2]
```diff
@@ -249,20 +249,20 @@ private int findBucketIndex(CatalogObjectRevision catalogObjectRevision) {
     private Page<CatalogObjectRevision> doQueryfindAllWorkflowRevisions(BucketMetadata bucket, long workflowId,
```
Comment (marcocast) [inline src/integration-test/java/org/ow2/proactive/catalog/rest/service/repository/QueryDslCatalogObjectRevisionRepositoryIntegrationTest.java]: workflow should be Object right?

File: src/integration-test/java/org/ow2/proactive/catalog/rest/service/repository/QueryDslCatalogObjectRevisionRepositoryIntegrationTest.java [hunk 3]
```diff
@@ -249,20 +249,20 @@ private int findBucketIndex(CatalogObjectRevision catalogObjectRevision) {
...
             Optional<BooleanExpression> booleanExpression) {
 
-        return queryDslWorkflowRevisionRepository.findAllWorkflowRevisions(bucket.id,
-                                                                           workflowId,
-                                                                           createAnyQueryExpression(booleanExpression),
-                                                                           new PageRequest(0,
-                                                                                           getTotalNumberOfWorkflowRevisions()));
+        return queryDslWorkflowRevisionRepository.findAllCatalogObjectRevisions(bucket.id,
+                                                                                workflowId,
```
Comment (marcocast) [inline src/integration-test/java/org/ow2/proactive/catalog/rest/service/repository/QueryDslCatalogObjectRevisionRepositoryIntegrationTest.java]: workflow should be Object right?

File: src/integration-test/java/org/ow2/proactive/catalog/rest/service/repository/QueryDslCatalogObjectRevisionRepositoryIntegrationTest.java [hunk 4]
```diff
@@ -249,20 +249,20 @@ private int findBucketIndex(CatalogObjectRevision catalogObjectRevision) {
...
             Optional<BooleanExpression> booleanExpression) {
 
-        return queryDslWorkflowRevisionRepository.findMostRecentWorkflowRevisions(bucket.id,
-                                                                                  createAnyQueryExpression(booleanExpression),
-                                                                                  new PageRequest(0,
-                                                                                                  getTotalNumberOfWorkflowRevisions()));
+        return queryDslWorkflowRevisionRepository.findMostRecentCatalogObjectRevisions(bucket.id,
+                                                                                       createAnyQueryExpression(booleanExpression),
+                                                                                       new PageRequest(0,
+                                                                                                       getTotalNumberOfWorkflowRevisions()));
```
Comment (marcocast) [inline src/integration-test/java/org/ow2/proactive/catalog/rest/service/repository/QueryDslCatalogObjectRevisionRepositoryIntegrationTest.java]: workflow should be Object right?

File: src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java [hunk 5]
```diff
@@ -70,53 +68,33 @@
...
-        }
+        return new CatalogObjectMetadataList(catalogService.createCatalogObject(kind,
+                                                                                name,
+                                                                                commitMessage,
+                                                                                bucketId,
+                                                                                layout,
+                                                                                file.getBytes()));
     }
 
     @ApiOperation(value = "Gets a workflow's metadata by IDs", notes = "Returns metadata associated to the latest revision of the workflow.")
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java]: workflow should be Object right?

File: src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java [hunk 6]
```diff
@@ -70,53 +68,33 @@
...
-                return ResponseEntity.badRequest().build();
-            }
-        }
+    @RequestMapping(value = "/buckets/{bucketId}/resources/{id}", method = GET)
+    public ResponseEntity<?> get(@PathVariable Long bucketId, @PathVariable Long id, HttpServletResponse response)
+            throws MalformedURLException {
+        return catalogService.getCatalogObjectMetadata(bucketId, id);
     }
 
     @ApiOperation(value = "Lists workflows metadata", notes = "Returns workflows metadata associated to the latest revision.")
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java]: workflow should be Object right?

File: src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java [hunk 7]
```diff
@@ -125,19 +103,19 @@ public CatalogObjectMetadataList create(@PathVariable Long bucketId,
...
                                                                                                                                   "Default sort order is ascending. " + "Multiple sort criteria are supported.") })
     @ApiResponses(value = @ApiResponse(code = 404, message = "Bucket not found"))
-    @RequestMapping(value = "/buckets/{bucketId}/workflows", method = GET)
+    @RequestMapping(value = "/buckets/{bucketId}/resources", method = GET)
     public PagedResources list(@PathVariable Long bucketId,
-            @ApiParam("Query string for searching workflows. See <a href=\"http://doc.activeeon.com/latest/user/ProActiveUserGuide.html#_searching_for_workflows\">Searching for workflows</a> for more information about supported attributes and operations.") @RequestParam(required = false) Optional<String> query,
+            @ApiParam("Query string for searching catalog objects. See <a href=\"http://doc.activeeon.com/latest/user/ProActiveUserGuide.html#_searching_for_workflows\">Searching for workflows</a> for more information about supported attributes and operations.") @RequestParam(required = false) Optional<String> query,
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java]: workflow should be Object right?

File: src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java [hunk 8]
```diff
@@ -125,19 +103,19 @@ public CatalogObjectMetadataList create(@PathVariable Long bucketId,
...
-        return workflowService.listWorkflows(bucketId, query, pageable, assembler);
+        return catalogService.listCatalogObjects(bucketId, query, pageable, assembler);
     }
 
-    @ApiOperation(value = "Delete a workflow", notes = "Delete the entire workflow as well as its revisions. Returns the deleted CatalogObject's metadata")
-    @ApiResponses(value = @ApiResponse(code = 404, message = "Bucket or workflow not found"))
-    @RequestMapping(value = "/buckets/{bucketId}/workflows/{workflowId}", method = DELETE)
+    @ApiOperation(value = "Delete a workflow", notes = "Delete the entire catalog object as well as its revisions. Returns the deleted CatalogObject's metadata")
+    @ApiResponses(value = @ApiResponse(code = 404, message = "Bucket or object not found"))
+    @RequestMapping(value = "/buckets/{bucketId}/resources/{workflowId}", method = DELETE)
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java]: workflow should be Object right?

File: src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java [hunk 9]
```diff
@@ -125,19 +103,19 @@ public CatalogObjectMetadataList create(@PathVariable Long bucketId,
...
     public ResponseEntity<?> delete(@PathVariable Long bucketId, @PathVariable Long workflowId) {
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java]: workflow should be Object right?

File: src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java [hunk 10]
```diff
@@ -125,19 +103,19 @@ public CatalogObjectMetadataList create(@PathVariable Long bucketId,
...
-        return workflowService.delete(bucketId, workflowId);
+        return catalogService.delete(bucketId, workflowId);
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java]: workflow should be Object right?

File: src/main/java/org/ow2/proactive/catalog/rest/service/BucketService.java [hunk 11]
```diff
@@ -1,7 +1,7 @@
 /*
  * ProActive Parallel Suite(TM):
  * The Open Source library for parallel and distributed
- * objects & Scheduling, Orchestration, Cloud Automation
+ * Workflows & Scheduling, Orchestration, Cloud Automation
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/catalog/rest/service/BucketService.java]: workflow should be Object right?

File: src/main/java/org/ow2/proactive/catalog/rest/service/CatalogObjectRevisionService.java [hunk 12]
```diff
@@ -291,7 +282,7 @@ private CatalogObjectRevision getWorkflowRevision(Long bucketId, Long workflowId
      * @param revisionId The revision number of the CatalogObject
      * @return The deleted CatalogObjectRevision metadata
      */
-    public ResponseEntity<?> delete(Long bucketId, Long workflowId, Optional<Long> revisionId) {
+    public ResponseEntity<CatalogObjectMetadata> delete(Long bucketId, Long workflowId, Optional<Long> revisionId) {
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/catalog/rest/service/CatalogObjectRevisionService.java]: workflow should be Object right?

File: src/main/java/org/ow2/proactive/catalog/rest/service/CatalogObjectRevisionService.java [hunk 13]
```diff
@@ -300,18 +291,18 @@ private CatalogObjectRevision getWorkflowRevision(Long bucketId, Long workflowId
                 catalogObjectRevision = (CatalogObjectRevision) iter.next();
                 CatalogObjectRevision newCatalogObjectRevisionReference = (CatalogObjectRevision) iter.next();
                 catalogObject.setLastCommitId(newCatalogObjectRevisionReference.getCommitId());
-                catalogObjectRevision = catalogObjectRevisionRepository.getWorkflowRevision(bucketId,
-                                                                                            workflowId,
-                                                                                            catalogObjectRevision.getCommitId());
+                catalogObjectRevision = catalogObjectRevisionRepository.getCatalogObjectRevision(bucketId,
+                                                                                                 workflowId,
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/catalog/rest/service/CatalogObjectRevisionService.java]: workflow should be Object right?

File: src/main/java/org/ow2/proactive/catalog/rest/service/CatalogObjectRevisionService.java [hunk 14]
```diff
@@ -300,18 +291,18 @@ private CatalogObjectRevision getWorkflowRevision(Long bucketId, Long workflowId
...
-                                                                                            workflowId,
-                                                                                            revisionId.get());
+                catalogObjectRevision = catalogObjectRevisionRepository.getCatalogObjectRevision(bucketId,
+                                                                                                 workflowId,
+                                                                                                 revisionId.get());
                 catalogObjectRevisionRepository.delete(catalogObjectRevision);
             }
         } else {
-            catalogObjectRevision = catalogObjectRepository.getMostRecentWorkflowRevision(bucketId, workflowId);
+            catalogObjectRevision = catalogObjectRepository.getMostRecentCatalogObjectRevision(bucketId, workflowId);
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/catalog/rest/service/CatalogObjectRevisionService.java]: workflow should be Object right?

File: src/main/java/org/ow2/proactive/catalog/rest/service/CatalogObjectRevisionService.java [hunk 15]
```diff
@@ -322,13 +313,9 @@ private CatalogObjectRevision getWorkflowRevision(Long bucketId, Long workflowId
     public Link createLink(Long bucketId, Long workflowId, CatalogObjectRevision catalogObjectRevision) {
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/catalog/rest/service/CatalogObjectRevisionService.java]: workflow should be Object right?

File: src/main/java/org/ow2/proactive/catalog/rest/service/CatalogObjectRevisionService.java [hunk 16]
```diff
@@ -322,13 +313,9 @@ private CatalogObjectRevision getWorkflowRevision(Long bucketId, Long workflowId
...
         ControllerLinkBuilder controllerLinkBuilder = linkTo(methodOn(CatalogObjectRevisionController.class).get(bucketId,
                                                                                                                  workflowId,
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/catalog/rest/service/CatalogObjectRevisionService.java]: workflow should be Object right?

File: src/main/java/org/ow2/proactive/catalog/rest/service/CatalogObjectService.java [hunk 17]
```diff
@@ -77,40 +71,17 @@ public CatalogObjectMetadata createCatalogObject(String kind, String name, Strin
...
-            return extractedWorkflows.stream()
-                                     .map(workflowFile -> createCatalogObject(kind,
-                                                                              name,
-                                                                              commitMessage,
-                                                                              bucketId,
-                                                                              layout,
-                                             proActiveWorkflowsArchive))
-                                     .collect(Collectors.toList());
-        }
+    public ResponseEntity<?> getCatalogObjectMetadata(long bucketId, long workflowId) {
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/catalog/rest/service/CatalogObjectService.java]: workflow should be Object right?

File: src/main/java/org/ow2/proactive/catalog/rest/service/CatalogObjectService.java [hunk 18]
```diff
@@ -77,40 +71,17 @@ public CatalogObjectMetadata createCatalogObject(String kind, String name, Strin
...
+        return catalogObjectRevisionService.getCatalogObject(bucketId, workflowId, Optional.empty());
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/catalog/rest/service/CatalogObjectService.java]: workflow should be Object right?

File: src/main/java/org/ow2/proactive/catalog/rest/service/CatalogObjectService.java [hunk 19]
```diff
@@ -77,40 +71,17 @@ public CatalogObjectMetadata createCatalogObject(String kind, String name, Strin
...
-    }
-
-    public PagedResources listWorkflows(Long bucketId, Optional<String> query, Pageable pageable,
+    public PagedResources listCatalogObjects(Long bucketId, Optional<String> query, Pageable pageable,
             PagedResourcesAssembler assembler) throws QueryExpressionBuilderException {
         return catalogObjectRevisionService.listCatalogObjects(bucketId, Optional.empty(), query, pageable, assembler);
     }
 
-    public ResponseEntity<?> delete(Long bucketId, Long workflowId) {
+    public ResponseEntity<CatalogObjectMetadata> delete(Long bucketId, Long workflowId) {
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/catalog/rest/service/CatalogObjectService.java]: workflow should be Object right?

File: src/main/java/org/ow2/proactive/catalog/rest/service/CatalogObjectService.java [hunk 20]
```diff
@@ -77,40 +71,17 @@ public CatalogObjectMetadata createCatalogObject(String kind, String name, Strin
...
         return catalogObjectRevisionService.delete(bucketId, workflowId, Optional.empty());
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/catalog/rest/service/CatalogObjectService.java]: workflow should be Object right?

File: src/test/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectControllerTest.java [hunk 21]
```diff
@@ -129,17 +129,17 @@ public void testList() throws Exception {
         when(pagedResourcesAssembler.toResource(any(Page.class),
                                                 any(CatalogObjectRevisionResourceAssembler.class))).thenReturn(mock(PagedResources.class));
         workflowController.list(1L, Optional.empty(), null, pagedResourcesAssembler);
-        verify(workflowService, times(1)).listWorkflows(anyLong(),
-                                                        any(Optional.class),
-                                                        any(Pageable.class),
-                                                        any(PagedResourcesAssembler.class));
+        verify(workflowService, times(1)).listCatalogObjects(anyLong(),
```
Comment (marcocast) [inline src/test/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectControllerTest.java]: workflow should be Object right?

File: src/test/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectControllerTest.java [hunk 22]
```diff
@@ -129,17 +129,17 @@ public void testList() throws Exception {
...
+                                                             any(Optional.class),
+                                                             any(Pageable.class),
+                                                             any(PagedResourcesAssembler.class));
     }
 
     @Test
     public void testGet() throws Exception {
         HttpServletResponse response = mock(HttpServletResponse.class);
         workflowController.get(1L, Collections.singletonList(2L), Optional.empty(), response);
```
Comment (marcocast) [inline src/test/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectControllerTest.java]: workflow should be Object right?

File: src/test/java/org/ow2/proactive/catalog/rest/service/CatalogObjectRevisionServiceTest.java [hunk 23]
```diff
@@ -222,14 +222,14 @@ public void testListWorkflowsWithoutWorkflowId() throws Exception {
     public void testGetWorkflowWithInvalidRevisionId() throws Exception {
         when(bucketRepository.findOne(anyLong())).thenReturn(mock(Bucket.class));
         when(workflowRepository.findOne(anyLong())).thenReturn(mock(CatalogObject.class));
```
Comment (marcocast) [inline src/test/java/org/ow2/proactive/catalog/rest/service/CatalogObjectRevisionServiceTest.java]: workflow should be Object right?

File: src/test/java/org/ow2/proactive/catalog/rest/service/CatalogObjectRevisionServiceTest.java [hunk 24]
```diff
@@ -222,14 +222,14 @@ public void testListWorkflowsWithoutWorkflowId() throws Exception {
...
-        when(workflowRepository.getMostRecentWorkflowRevision(anyLong(), anyLong())).thenReturn(null);
+        when(workflowRepository.getMostRecentCatalogObjectRevision(anyLong(), anyLong())).thenReturn(null);
         workflowRevisionService.getCatalogObject(DUMMY_ID, DUMMY_ID, Optional.empty(), Optional.empty());
     }
 
     @Test
     public void testGetWorkflowWithValidRevisionIdNoPayload() throws Exception {
         getWorkflow(Optional.of(DUMMY_ID), Optional.empty());
```
Comment (marcocast) [inline src/test/java/org/ow2/proactive/catalog/rest/service/CatalogObjectRevisionServiceTest.java]: workflow should be Object right?

File: src/test/java/org/ow2/proactive/catalog/rest/service/CatalogObjectRevisionServiceTest.java [hunk 25]
```diff
@@ -252,9 +252,11 @@ private void getWorkflow(Optional<Long> revisionId, Optional<String> alt) throws
         when(workflowRepository.findOne(anyLong())).thenReturn(mock(CatalogObject.class));
```
Comment (marcocast) [inline src/test/java/org/ow2/proactive/catalog/rest/service/CatalogObjectRevisionServiceTest.java]: workflow should be Object right?

File: src/main/java/org/ow2/proactive/catalog/rest/service/BucketService.java (hunk 11 above)
Comment (laurianed) [inline src/main/java/org/ow2/proactive/catalog/rest/service/BucketService.java]: Not here lol

**Pull request merged**